presto-main/src/main/java/com/facebook/presto/execution/NodeTaskMap.java

```diff
@@ -108,87 +117,160 @@ private void addTask(RemoteTask task)
+                    totalValue.addAndGet(-oldValue);
+                    throw new IllegalArgumentException(stat + " is negative");
                 }
 
-                int oldValue = localPartitionedSplitCount.getAndSet(partitionedSplitCount);
-                nodeTotalPartitionedSplitCount.addAndGet(partitionedSplitCount - oldValue);
+                long oldValue = this.value.getAndSet(value);
+                totalValue.addAndGet(value - oldValue);
             }
 
             public void cleanup()
             {
-                int leakedSplits = localPartitionedSplitCount.getAndSet(0);
-                if (leakedSplits == 0) {
+                long leakedValues = value.getAndSet(0);
+                if (leakedValues == 0) {
                     return;
                 }
 
-                log.error("BUG! %s for %s leaked with %s partitioned splits.  Cleaning up so server can continue to function.",
+                log.error("BUG! %s for %s leaked with %s %s.  Cleaning up so server can continue to function.",
                         getClass().getName(),
                         taskId,
-                        leakedSplits);
+                        leakedValues,
+                        stat);
 
-                nodeTotalPartitionedSplitCount.addAndGet(-leakedSplits);
+                totalValue.addAndGet(-leakedValues);
             }
 
             @Override
             public String toString()
             {
                 return toStringHelper(this)
                         .add("taskId", taskId)
-                        .add("splits", localPartitionedSplitCount)
+                        .add(stat, value)
                         .toString();
             }
         }
+
+        // tracks stats which are passed as accumulated (cpu time) by calculating delta / duration.
+        @ThreadSafe
+        private class AccumulatedTaskStatsTracker
+                extends TaskStatsTracker
+        {
+            private long previousTaskAge;
+            private long previousValue;
+
+            AccumulatedTaskStatsTracker(String stat, TaskId taskId, AtomicLong totalValue)
+            {
+                super(stat, taskId, totalValue);
+            }
+
+            private long getDeltaPerSecond(long taskAgeInMillis, long value)
+            {
+                if (previousTaskAge == 0 && value > 0) {
+                    previousTaskAge = taskAgeInMillis;
+                    previousValue = value;
+                    return 0;
+                }
+
+                if (taskAgeInMillis <= previousTaskAge) {
+                    return 0;
+                }
+
+                if (value > 0) {
+                    long deltaValue = (value - previousValue) * 100;
+                    long deltaDuration = taskAgeInMillis - previousTaskAge;
+                    previousTaskAge = taskAgeInMillis;
+                    previousValue = value;
+                    return deltaValue > 0 ? deltaValue / deltaDuration : 0;
```

Comment:
Have you checked that we're not losing anything by using whole number arithmetic?

Reply:
Converted to a double.